pilot/pkg/model/context.go

```diff
@@ -102,6 +102,15 @@ func (node *Proxy) ServiceNode() string {
 
 }
 
+// Has10Config returns true if the sidecar has at least 1.0 config.
+// This will be removed after 1.0, and possibly replaced with a more fine-grained check to verify
+// sidecar capabilities. It is used to detect 0.8 sidecards and allow upgrades without breaking 0.8 workloads during
+// upgrade.
+func (node *Proxy) Has10Config() bool {
+	_, found := node.Metadata["ISTIO_PROXY_VERSION"]
+	return found
+}
```

Comment:
If you are going to move this to the model, then please change this to `GetProxyVersion() string` where for 0.8, you return 0.8, and for 1.0, you return 1.0 based on presence of the ISTIO_PROXY_VERSION metadata. I do not want a has10config, has11config, has13config, and keep having to remove it from every point of reference after a release.

Reply:
I want to hear what @costinm has to say here. It's his code.

Reply:
Also, what you are suggesting here @rshriram is strictly additive. Mixerplugin only cares if it's 1.0 or greater, which is what this function checks for. If there is a need to detect 1.1 or whatever, then we can just add more functions.

Reply:
I think what @rshriram is asking is to return the value of node.Metadata["ISTIO_PROXY_VERSION"] and make this function more generic.

And then the caller can make the decision based on the version. (We may eventually need a util like isBefore(v1, v2) or isAfter(v1, v2) somewhere).

Reply:
Done